### PR TITLE
Dark mode - Frontend

### PR DESCRIPTION
**Pull request by AdrianAndersen**
Adds the option to turn on Dark Theme.

image
image

### CODE REVIEW

File: app/components/Upload/UploadImage.css [hunk 1]
```diff
@@ -48,7 +54,7 @@
 }
 
 .placeholdeTitle {
-  background: rgba(255, 255, 255, 0.8);
+  background: var(--color-white);
```
Comment (LudvigHz) [inline app/components/Upload/UploadImage.css]: Make sure this doesn't change anything

File: app/routes/bdb/components/bdb.css [hunk 2]
```diff
@@ -134,11 +134,11 @@
 }
 
 .contacted {
-  background-color: #ffff7f;
+  background-color: var(--color-white);
```
Comment (LudvigHz) [inline app/routes/bdb/components/bdb.css]: Here as well

File: app/routes/surveys/utils.js [hunk 3]
```diff
@@ -82,10 +82,10 @@ export const defaultActiveFrom = (hours: number, minutes: number) =>
...
-  '#2b95d6',
-  '#d9822b',
-  '#3dcc91',
+  'var(--lego-link-color)',
+  'var(--color-blue-4)',
+  'var(--color-orange-3)',
+  'var(--color-green-5)',
   '#c73aea',
   '#f4ee42',
   '#98f442',
```
Comment (LudvigHz) [inline app/routes/surveys/utils.js]: Why are only some of these edited?

File: app/styles/icomoon.css [hunk 4]
```diff
@@ -28,7 +28,7 @@
   text-transform: none;
   line-height: 1;
   -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
+  -moz-osx-font-smoothing: var(--color-gray-1);
```
Comment (LudvigHz) [inline app/styles/icomoon.css]: Does this work?

File: app/components/Footer/Footer.css [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 .footer {
   width: 100%;
-  background-color: #1b1a1a;
-  color: white;
+  background-color: var(--color-black);
```
[the remark this answers is not in the document]
Comment (AdrianAndersen) [reply app/components/Footer/Footer.css]: Hi. Thanks for the review. I will look into the colors you specified. I agree that this sould be its own variable.

File: app/components/Form/CheckBox.css [hunk 6]
```diff
@@ -30,14 +30,14 @@
...
-  color: white;
+  color: var(--color-white);
   transition: all 0.2s;
   opacity: 1;
   transform: scale(1);
 }
 
 .unchecked + label::before {
-  background: #f8f8f8;
+  background: var(--color-white);
```
[the remark this answers is not in the document]
Comment (AdrianAndersen) [reply app/components/Form/CheckBox.css]: Added a varable set with --almost-white-x, to make this look good

File: app/components/Form/TextInput.css [hunk 7]
```diff
@@ -1,30 +1,36 @@
...
-  border: 1px solid #ccc;
+  border: 1px solid var(--color-mono-gray-3);
   border-radius: 3px;
   width: 100%;
+  background-color: inherit;
 }
 
 .disabled {
-  background-color: #f9f9f9;
+  background-color: var(--color-white);
```
[the remark this answers is not in the document]
Comment (AdrianAndersen) [reply app/components/Form/TextInput.css]: Moved to --almost-white-1 (#f9f9f9)

File: app/components/Modal/ConfirmModal.css [hunk 8]
```diff
@@ -9,7 +9,7 @@
 
 .backdrop {
   position: fixed;
-  background: rgba(0, 0, 0, 0.8);
+  background: var(--color-black);
```
[the remark this answers is not in the document]
Comment (AdrianAndersen) [reply app/components/Modal/ConfirmModal.css]: Moved to the original color, using rgba-variables

File: app/components/Upload/UploadImage.css (hunk 1 above)
Comment (AdrianAndersen) [reply app/components/Upload/UploadImage.css]: Moved to the original color, using rgba-variables

File: app/routes/bdb/components/bdb.css (hunk 2 above)
Comment (AdrianAndersen) [reply app/routes/bdb/components/bdb.css]: Reverted, as this color should not be changed in darkMode

File: app/routes/surveys/utils.js (hunk 3 above)
Comment (AdrianAndersen) [reply app/routes/surveys/utils.js]: Here, some of the colors were already in the variable sheet. I have now created new variables for the rest of the colors, and applied them here.

File: app/styles/icomoon.css (hunk 4 above)
Comment (AdrianAndersen) [reply app/styles/icomoon.css]: No, this does not work, as the variable sheet is not imported. Revertd back to grayscale asdoes not impact darkMode in any way.

File: app/components/Header/index.js [hunk 9]
```diff
@@ -109,9 +112,22 @@ class Header extends Component<Props, State> {
...
     let title, form;
+
+    // This should be optimized, so that it does not apply theme every time Header renders,
+    // but rather when a theme change happens
+
+    // Using hooks?
+    /*useEffect(() => {
+      applySelectedTheme((user && user.selectedTheme) || 'light');
+    }, [loggedIn, user && user.selectedTheme]);*/
+
```
[the remark this answers is not in the document]
Comment (AdrianAndersen) [reply app/components/Header/index.js]: Resolved this with a simple if-test, that checks if loggedin && user && user.selectedTheme != the applied theme.

**Pull request merged**